lib/Helper/MergeConfig.ts

```diff
@@ -0,0 +1,185 @@
+import * as fs from 'fs';
+import * as ts from 'typescript';
+
+// checks if statement uses module.exports format
+function isModuleExport(statement: ts.ExpressionStatement): boolean {
+  if (statement.kind !== ts.SyntaxKind.ExpressionStatement) {
+    return false;
+  }
+  if (!statement.expression) {
+    return false;
+  }
+  const { left } = statement.expression;
+  // check if this is module.exports
+  return (
+    left.expression?.escapedText === 'module' &&
+    left.name?.escapedText === 'exports'
+  );
+}
+
+// checks if statement uses default export format
+function isDefaultExport(statement: ts.Statement): boolean {
+  if (statement.kind !== ts.SyntaxKind.ExportAssignment) {
+    return false;
+  }
+  if (!statement.expression) {
+    return false;
+  }
+  return true;
+}
+
+// finds the the source variable name from the statement
+function getSourceVariableName(
+  statement: ts.Statement,
+  sourceText: string,
+): { variableName: string | undefined; text: string | undefined } {
+  let sourceVariableName, declarationText;
+  if (isModuleExport(statement)) {
+    const { right } = statement.expression;
+
+    // modules.exports = () => ({...})
+    if (
+      right?.kind === ts.SyntaxKind.ArrowFunction ||
+      right?.kind === ts.SyntaxKind.FunctionExpression
+    ) {
+      ({ sourceVariableName, declarationText } = parseFunction(
+        statement,
+        sourceText,
+      ));
+    }
+    // module.exports = {...}
+    else {
+      sourceVariableName = right?.escapedText;
+    }
+  }
+  // default export {...}
```

Comment:
@roggenkemper these kind of comments about which syntax you're looking for are great! Having them in the other methods would make this much easier to understand